pkg/registry/generic/matcher.go

```diff
@@ -93,9 +113,10 @@ type Matcher interface {
 	// include the object's namespace.
 	MatchesSingle() (key string, matchesSingleObject bool)
 
-	// TODO: when we start indexing objects, add something like the below:
-	//         MatchesIndices() (indexName []string, indexValue []string)
-	//       where indexName/indexValue are the same length.
+	// If the index with a given name is supported by the matcher,
+	// returns the value for that index and true, otherwise returns
+	// "" and false.
+	Index(indexName string) (indexValue string, indexSupported bool)
```

Comment:
Why do you pass in the index name? won't this require 1 call per index?

I like the suggestion in the TODO, because that will allow the caching layer to discover what things it could possibly build indexes for.